triton-vm/src/table/master_table.rs

```diff
@@ -284,35 +284,69 @@ where
         let evaluation_domain = self.evaluation_domain();
         let randomized_trace_domain = self.randomized_trace_domain();
         let num_rows = evaluation_domain.length;
+
+        profiler!(start "polynomial zero-initialization");
         let mut interpolation_polynomials = Array1::zeros(Self::NUM_COLUMNS);
+        profiler!(stop "polynomial zero-initialization");
 
         // compute interpolants
+        profiler!(start "interpolation");
         Zip::from(self.randomized_trace_table().axis_iter(Axis(1)))
             .and(interpolation_polynomials.axis_iter_mut(Axis(0)))
             .par_for_each(|trace_column, poly| {
                 let trace_column = trace_column.as_slice().unwrap();
                 let interpolation_polynomial = randomized_trace_domain.interpolate(trace_column);
                 Array0::from_elem((), interpolation_polynomial).move_into(poly);
             });
+        profiler!(stop "interpolation");
 
-        let mut extended_trace = Vec::<FF>::new();
+        let mut extended_trace = Vec::<FF>::with_capacity(0);
         let num_elements = num_rows * Self::NUM_COLUMNS;
-        let should_cache = crate::config::cache_lde_trace().map_or_else(
-            || extended_trace.try_reserve_exact(num_elements).is_ok(),
-            |cd| cd == CacheDecision::Cache,
-        );
+        let should_cache = match crate::config::cache_lde_trace() {
+            Some(CacheDecision::Cache) => {
+                extended_trace = Vec::with_capacity(num_elements);
+                true
+            }
+            None => extended_trace.try_reserve_exact(num_elements).is_ok(),
+            Some(CacheDecision::NoCache) => false,
+        };
 
         if should_cache {
-            extended_trace.resize(num_elements, FF::zero());
+            profiler!(start "resize");
+            unsafe {
+                // This is done to avoid having to write a value to each element
+                // in the vector which, when done sequentially, takes as long
+                // time as the evaluation itself. This is safe because:
+                // SAFETY:
+                // 1. The length is set to the same as the capacity.
+                // 2. Each element is over-written before being read in the
+                //    immediately following lines.
+                // 3. If we execute this unsafe code, then either:
+                //     - `Vec::with_capacity` was called; or
+                //     - `try_reserve_exact` was called *and returned
+                //       successfully*.
+                //    In either case, allocation happened.
```

Comment:
I think this comment is slightly outdated and can be condensed.

```suggestion
                // Speed up initialization through parallelization.
                //
                // SAFETY:
                // 1. The capacity is sufficiently large – see above `assert!`.
                // 2. The length is set to equal (or less than) the capacity.
                // 3. Each element is over-written before being read in the
                //    immediately following lines.
```